lib/version_win32.c

```diff
@@ -214,10 +226,23 @@ bool curlx_verify_windows_version(const unsigned int majorVersion,
   if(platform != PLATFORM_DONT_CARE)
     cm = VerSetConditionMask(cm, VER_PLATFORMID, VER_EQUAL);
 
-  if(VerifyVersionInfo(&osver, (VER_MAJORVERSION | VER_MINORVERSION |
-                                VER_SERVICEPACKMAJOR | VER_SERVICEPACKMINOR),
-                       cm))
-    matched = TRUE;
+  /* Later versions of Windows have version functions that may not return the
+     real version of Windows unless the application is so manifested. We prefer
+     the real version always, so we use the Rtl variant of the function when
+     possible. Note though the function signature and underlying fundamental
+     types are the same, the return values are different. */
+  if(pRtlVerifyVersionInfo) {
+    matched = !pRtlVerifyVersionInfo(&osver,
+      (VER_MAJORVERSION | VER_MINORVERSION |
+       VER_SERVICEPACKMAJOR | VER_SERVICEPACKMINOR),
+      cm);
+  }
+  else {
+    matched = !!VerifyVersionInfoW(&osver,
```

Comment:
Why double negation here?

Reply:
VerifyVersionInfoW returns BOOL nonzero for success which is, technically in MS world, not guaranteed to be 1. In order for matched to return a guaranteed bool true/false (ie 1/0) I did a double negation. I could instead do it like `else if(VerifyVersionInfoW... {matched = true} else matched = false;` but I think the double negation is cleaner